### PR TITLE
feat(client): validator key consistency check once per epoch

### PR DESCRIPTION
**Pull request by darioush**
- At key file load time (`KeyFile::from_file`), verify the private key derives the same public key listed in the file. Returns `InvalidData` error on mismatch.
- At each epoch transition, verify the loaded validator public key matches the key registered in the upcoming epoch's info. Panics on mismatch with an actionable message. The check is skipped during sync to avoid false positives when the node is behind and the key was correctly rotated after the local head's epoch.
- The check fires on the first non-syncing head block of each epoch (via `check_validator_key_on_new_head`), so it also catches the case where the epoch boundary was crossed during sync.
- Add `validator_signer` field to `NodeSetupState` to support injecting a custom signer for testing.
- Add unit test for key file mismatch and a test-loop test for epoch boundary mismatch detection.

### CODE REVIEW

File: test-loop-tests/src/tests/validator_key_check.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#[should_panic(expected = "validator key mismatch")]
+fn test_validator_key_mismatch_at_epoch_boundary() {
+    init_test_logger();
+
+    let epoch_length = 5;
+    let mut env = TestLoopBuilder::new().validators(4, 0).epoch_length(epoch_length).build();
+
+    // Run most of the first epoch (stop 1 block before boundary).
+    env.validator_runner().run_until_head_height(epoch_length - 1);
+
```
[the remark this answers is not in the document]
Comment (darioush) [reply test-loop-tests/src/tests/validator_key_check.rs]: I think it works out because genesis_height is default 1.

File: chain/client/src/client.rs [hunk 2]
```diff
@@ -1654,6 +1680,12 @@ impl Client {
             // TODO make sure transactions don't get added for the old shard
             // layout after the pool resharding
             if self.epoch_manager.is_next_block_epoch_start(&block_hash).unwrap_or(false) {
+                if let Ok(next_epoch_id) =
+                    self.epoch_manager.get_epoch_id_from_prev_block(&block_hash)
+                {
+                    self.check_validator_key_for_epoch(&next_epoch_id);
```
[the remark this answers is not in the document]
Comment (darioush) [reply chain/client/src/client.rs]: Seems not worth it to me, can add it if reviewer prefers to have it.

File: chain/client/src/client_actor.rs [hunk 3]
```diff
@@ -436,6 +436,8 @@ impl ClientActor {
         if let Some(vs) = &client.validator_signer.get() {
             tracing::info!(target: "client", validator_id = %vs.validator_id(), "starting validator node");
             check_validator_tracked_shards(&client, vs.validator_id())?;
+            let epoch_id = client.chain.head()?.epoch_id;
+            client.check_validator_key_for_epoch(&epoch_id);
```
Comment (wacban) [inline chain/client/src/client_actor.rs]: Is there a case where the node is behind and wrong epoch is used for the check?

Comment (darioush) [reply chain/client/src/client_actor.rs]: Great point. Let me figure it out before proceeding.

File: chain/client/src/client.rs [hunk 4]
```diff
@@ -181,6 +181,10 @@ pub struct Client {
     chunk_producer_accounts_cache: Option<(EpochId, Arc<Vec<AccountId>>)>,
     /// Reed-Solomon encoder for shadow chunk validation.
     shadow_validation_reed_solomon: OnceLock<Arc<ReedSolomon>>,
+    /// The last epoch for which the validator key was checked. Used to ensure
+    /// the check runs once per epoch on the first non-syncing head block, even
+    /// if the epoch boundary was crossed during sync.
+    last_validator_key_check_epoch: Option<EpochId>,
```
Comment (darioush) [inline chain/client/src/client.rs:187]: I'm also happy to just check it at the beginning of an epoch to avoid adding this field to client

File: chain/client/src/client.rs [hunk 5]
```diff
@@ -210,6 +214,43 @@ impl Client {
...
+        if self.sync_handler.sync_status.is_syncing() {
+            return;
+        }
+        let Some(signer) = self.validator_signer.get() else { return };
+        let epoch_id = block.header().epoch_id();
+        if self.last_validator_key_check_epoch.as_ref() == Some(epoch_id) {
+            return;
+        }
+        let validator_id = signer.validator_id();
+        let epoch_info = match self.epoch_manager.get_epoch_info(epoch_id) {
```
Comment (wacban) [inline chain/client/src/client.rs]: nit: you can also use the `let Ok(epoch_info) = ... else { tracing::error!(...); };` pattern here

File: chain/client/src/client.rs [hunk 6]
```diff
@@ -210,6 +214,43 @@ impl Client {
...
+                tracing::error!(target: "client", ?epoch_id, ?err, "failed to get epoch info for validator key check");
+                return;
+            }
+        };
+        self.last_validator_key_check_epoch = Some(*epoch_id);
+        let Some(validator_stake) = epoch_info.get_validator_by_account(validator_id) else {
+            return;
+        };
+        let local_key = signer.public_key();
+        let epoch_key = validator_stake.public_key();
```
Comment (wacban) [inline chain/client/src/client.rs]: epoch_key doesn't quite convince me, how about chain_key or on_chain_key?

**Comment by darioush** [comment]
Making https://github.com/near/nearcore/issues/15469 to track the validator key rotation issue.
